reporter/iface.go

```diff
@@ -36,46 +35,28 @@ type TraceReporter interface {
 	ReportTraceEvent(trace *libpf.Trace, meta *samples.TraceEventMeta) error
 }
 
-// ExecutableOpener is a function that attempts to open an executable.
-type ExecutableOpener = func() (process.ReadAtCloser, error)
+type ExecutableMetadata struct {
+	// MappingFile is the reference to mapping file data.
+	MappingFile libpf.FrameMappingFile
+
+	// Process is the interface to the process holding the file.
+	Process process.Process
+
+	// Mapping is the process.Mapping file. Process.OpenMappingFile can be used
+	// to open the file if needed.
+	Mapping *process.Mapping
```

Comment:
I think, we can remove them. They are set in `interpreter/dotnet` and `processmanager`, but never used as far as I can tell.
 
```suggestion
```

Reply:
They are currently used by external projects as a way to hook into executable detection and upload symbols see https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/55